oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -501,10 +535,18 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpCredentialsAdapter adapter = new HttpCredentialsAdapter(sourceCredentials);
     HttpRequestFactory requestFactory = httpTransport.createRequestFactory();
 
-    String endpointUrl =
-        this.iamEndpointOverride != null
-            ? this.iamEndpointOverride
-            : String.format(IAM_ACCESS_TOKEN_ENDPOINT, this.targetPrincipal);
+    String endpointUrl = null;
+    try {
+      endpointUrl =
+          this.iamEndpointOverride != null
+              ? this.iamEndpointOverride
+              : String.format(
+                  OAuth2Utils.IAM_ACCESS_TOKEN_ENDPOINT_FORMAT,
+                  getUniverseDomain(),
+                  this.targetPrincipal);
+    } catch (IOException e) {
+      throw new IOException("Error getting Universe Domain", e);
```

Comment:
This re-throw does not seem to provide much additional value, can we just let the `IOException` bubble up without wrapping and rethrowing?